### DISCUSSION TITLE
Appending to GITHUB_PATH and GITHUB_ENV

### DISCUSSION THREAD

**Discussion by dudicoco** [jdx/mise#3402] [Q&A] [answered]
I'm looking for a way to append all of the env vars defined within `.mise.toml` to [`GITHUB_PATH`](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-system-path) and [`GITHUB_ENV`](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable)

Is there a way to achieve this?

**Comment by jdx** [comment]
I think this should work:

```bash
mise env | grep -v "^export PATH" >> "$GITHUB_ENV"
mise bin-paths >> "$GITHUB_PATH"
```

**Reply by dudicoco** [reply]
Thanks, that got me almost all the way there.

I'm missing the env path definitions:
```toml
[env]
_.path = "./node_modules/.bin"
_.python.venv = { path = ".venv", create = true }
```

Which add to the path `.venv/bin` and `node_modules/.bin`.

**Reply by jdx** [reply]
`mise env | grep "^export PATH"` will give you the full PATH but it will also include things that were already in PATH. I don't know if we have a way to export only new PATH entries. We do in the codebase so it would be possible to add an option to an existing command or something to include that though. I don't know if including everything from PATH is a problem, but if nothing else you should be able to use something like `tr` to remove existing entries.

**Reply by dudicoco** [reply, marked as answer]
ended up with this
```sh
mise env | sed 's/^export//' | xargs -I {} echo "{}" >>"$GITHUB_ENV"
```
`"$GITHUB_PATH"` was not sufficient as the runner's PATH took precedence, so my venv bins (yamllint) were overriden with the runner's bins.
